File: library/src/main/java/com/nasahapps/nasahutils/app/BaseFragment.java
package com.nasahapps.nasahutils.app;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.Menu;
import android.view.MenuInflater;

import rx.Observable;
import rx.android.observables.AndroidObservable;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by Hakeem on 7/26/15.
 */
public abstract class BaseFragment extends Fragment {

    protected String TAG;
    protected Menu mMenu;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TAG = getClass().getSimpleName();
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        mMenu = menu;
    }

    public <T> Observable<T> getObservable(Observable<T> observable) {
        return AndroidObservable.bindFragment(this, observable)
                .subscribeOn(Schedulers.newThread())
                .observeOn(AndroidSchedulers.mainThread());
    }
}

File: library/src/main/java/com/nasahapps/nasahutils/app/BaseFragment.kt
package com.nasahapps.nasahutils.app

import androidx.fragment.app.Fragment
import android.view.Menu
import android.view.MenuInflater

/**
 * Created by Hakeem on 7/26/15.
 */
abstract class BaseFragment : Fragment() {

    protected val TAG = javaClass.simpleName
    protected var menu: Menu? = null

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        super.onCreateOptionsMenu(menu, inflater)
        this.menu = menu
    }
}
